plugins/TimingTriggerCandidateMaker.cpp

```diff
@@ -37,48 +38,87 @@ TimingTriggerCandidateMaker::TimingTriggerCandidateMaker(const std::string& name
   register_command("scrap", &TimingTriggerCandidateMaker::do_scrap);
 }
 
-triggeralgs::TriggerCandidate
+template<typename T> 
+std::vector<int> 
+TimingTriggerCandidateMaker::GetTriggeredBits(T signal_map)
+{
+  std::vector<int> triggered_bits;
+  int nbits = std::numeric_limits<T>::digits;
+  for (int i = 0; i < nbits; ++i) {
+    if ((signal_map >> i) & 1 ) {
+      triggered_bits.push_back(i);
+    }
+  }
+  return triggered_bits;
+}
+
+std::vector<triggeralgs::TriggerCandidate>
 TimingTriggerCandidateMaker::HSIEventToTriggerCandidate(const dfmessages::HSIEvent& data)
 {
-  triggeralgs::TriggerCandidate candidate;
+  std::vector<triggeralgs::TriggerCandidate> candidates;
   // TODO Trigger Team <dune-daq@github.com> Nov-18-2021: the signal field ia now a signal bit map, rather than unique
   // value -> change logic of below?
-  if (m_hsi_passthrough == true) {
-    TLOG_DEBUG(TLVL_DEBUG_MEDIUM) << "[TTCM] HSI passthrough applied, modified readout window is set";
-    candidate.time_start = data.timestamp - m_hsi_pt_before;
-    candidate.time_end = data.timestamp + m_hsi_pt_after;
-  } else {
-    if (m_detid_offsets_map.count(data.signal_map)) {
-      // clang-format off
-      candidate.time_start = data.timestamp - m_detid_offsets_map[data.signal_map].first;  // time_start
-      candidate.time_end   = data.timestamp + m_detid_offsets_map[data.signal_map].second; // time_end,
-      // clang-format on    
-    } else {
+
+  std::vector<int> triggered_bits = GetTriggeredBits(data.signal_map);
+  for (int bit_index : triggered_bits) {
+    uint32_t signal = 1 << bit_index;
+
+    if (!m_hsisignal_map.count(signal)) {
       throw dunedaq::trigger::SignalTypeError(ERS_HERE, get_name(), data.signal_map);
     }
-  }
-  candidate.time_candidate = data.timestamp;
-  // throw away bits 31-16 of header, that's OK for now
-  candidate.detid = { static_cast<triggeralgs::detid_t>(data.signal_map) }; // NOLINT(build/unsigned)
-  candidate.type = triggeralgs::TriggerCandidate::Type::kTiming;
 
-  candidate.algorithm = triggeralgs::TriggerCandidate::Algorithm::kHSIEventToTriggerCandidate;
-  candidate.inputs = {};
+    triggeralgs::TriggerCandidate candidate;
+    candidate.time_start = data.timestamp - m_hsisignal_map[data.signal_map].time_before;  // time_start
+    candidate.time_end   = data.timestamp + m_hsisignal_map[data.signal_map].time_after; // time_end,
+    candidate.time_candidate = data.timestamp;
+    // throw away bits 31-16 of header, that's OK for now
+    candidate.detid = { static_cast<triggeralgs::detid_t>(signal) }; // NOLINT(build/unsigned)
```

Comment:
Is this still needed (storing data to `detid` that is not detid) ?

Reply:
Probably not, I just didn't want to make this change here. I can change to header instead?

Reply:
I think the idea was to drop passing this information through. We are using the header in other places - although I'm not sure that is used/useful.